Cannot use plain text in @OpenShiftResource

##### Issue Overview

The JavaDoc says that it it possible to define the resource that should be deployed in OpenShift in the `@OpenShiftResource` annotation directly, but it's not true and creation of the resource fails.

##### Expected Behaviour

Resource is deployed.

##### Current Behaviour

```
	at org.arquillian.cube.openshift.impl.resources.OpenShiftResourceFactory.createResources(OpenShiftResourceFactory.java:76)
	at org.arquillian.cube.openshift.impl.CEEnvironmentProcessor.createEnvironment(CEEnvironmentProcessor.java:91)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeClass(EventTestRunnerAdaptor.java:89)
	at org.jboss.arquillian.junit.Arquillian$2.evaluate(Arquillian.java:163)
	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:350)
	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:54)
	at org.jboss.arquillian.junit.Arquillian$3.evaluate(Arquillian.java:177)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:115)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:367)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:274)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:161)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
Caused by: java.lang.IllegalArgumentException: java.net.MalformedURLException: no protocol: {
  "apiVersion": "v1",
  "kind": "ServiceAccount",
  "metadata": {
    "name": "eap-service-account"
  }
}
	at org.arquillian.cube.kubernetes.impl.resolver.ResourceResolver.resolve(ResourceResolver.java:30)
	at org.arquillian.cube.openshift.impl.resources.OpenShiftResourceFactory.createOpenShiftResources(OpenShiftResourceFactory.java:116)
	at org.arquillian.cube.openshift.impl.resources.OpenShiftResourceFactory.createResources(OpenShiftResourceFactory.java:66)
	... 38 more
Caused by: java.net.MalformedURLException: no protocol: {
  "apiVersion": "v1",
  "kind": "ServiceAccount",
  "metadata": {
    "name": "eap-service-account"
  }
}
	at java.net.URL.<init>(URL.java:593)
	at java.net.URL.<init>(URL.java:490)
	at java.net.URL.<init>(URL.java:439)
	at org.arquillian.cube.kubernetes.impl.resolver.ResourceResolver.resolve(ResourceResolver.java:27)
	... 40 more
```

--- a/kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/resolver/ResourceResolver.java
+++ b/kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/resolver/ResourceResolver.java
@@ -1,5 +1,8 @@
 package org.arquillian.cube.kubernetes.impl.resolver;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
 import java.io.IOException;
 import java.net.URL;
 
@@ -9,12 +12,29 @@ public class ResourceResolver {
     static final String FILE_PREFIX = "file";
     static final String CLASSPATH_PREFIX = "classpath:";
 
-    public static URL resolve(String location) {
+    private static File createTemporalDefinition(String content) throws IOException {
+        // In case it is not a http, file nor a classpath protocol,
+        // we assume that this is plain text. We store it in a temporary
+        // file and return the URL to it.
+        File tmp = File.createTempFile("arquillian-cube", ".res");
+
+        // Remove the temporary file after running the test
+        tmp.deleteOnExit();
+
+        // Write content to temporary file
+        BufferedWriter writer = new BufferedWriter(new FileWriter(tmp));
+        writer.write(content);
+        writer.close();
+
+        return tmp;
+    }
+
+    public static URL resolve(String content) {
         try {
-            if (location.startsWith(URL_PREFIX) || location.startsWith(FILE_PREFIX)) {
-                return new URL(location);
-            } else if (location.startsWith(CLASSPATH_PREFIX)) {
-                String classPathLocation = location.substring(location.indexOf(CLASSPATH_PREFIX)
+            if (content.startsWith(URL_PREFIX) || content.startsWith(FILE_PREFIX)) {
+                return new URL(content);
+            } else if (content.startsWith(CLASSPATH_PREFIX)) {
+                String classPathLocation = content.substring(content.indexOf(CLASSPATH_PREFIX)
                     + CLASSPATH_PREFIX.length());
                 final URL resource = Thread.currentThread().getContextClassLoader().getResource(classPathLocation);
 
@@ -24,7 +44,8 @@ public class ResourceResolver {
 
                 return resource;
             } else {
-                return new URL(location);
+                File tmp = ResourceResolver.createTemporalDefinition(content);
+                return tmp.toURI().toURL();
             }
         } catch (IOException e) {
             throw new IllegalArgumentException(e);

